sdk/monitor/opentelemetry-exporter-azure-monitor/test/common/flushSpanProcessor.ts

```diff
@@ -10,10 +10,11 @@ export class FlushSpanProcessor implements SpanProcessor {
   private _spans: ReadableSpan[] = [];
   constructor(public exporter: SpanExporter) {}
 
-  forceFlush(callback: () => void): void {
-    this.exporter.export(this._spans, () => {
-      this._spans = [];
-      callback();
+  forceFlush(): Promise<void> {
+    return Promise.resolve().then(() => {
+      this.exporter.export(this._spans, () => {
+        this._spans = [];
+      });
```

Comment:
This doesn't actually wait for export to complete before returning unless you change it up a bit;

```suggestion
    return new Promise((resolve) => {
      this.exporter.export(this._spans, () => {
        this._spans = [];
        resolve();
      });
```